### ISSUE TITLE
Sonification for current in the electromagnet 

### ISSUE THREAD

**Issue by Ashton-Morris** [phetsims/faradays-electromagnetic-lab#80]
I have a short video mockup here. [felCurrentLoopMockupVideo.mp4](https://o365coloradoedu.sharepoint.com/sites/PHYS-phet-pilot/_layouts/15/stream.aspx?id=%2Fsites%2FPHYS%2Dphet%2Dpilot%2FPhET%20Files%2FPhET%20Files%2FPhET%20Sim%20Design%2FFaraday%27s%20Electromagnetic%20Lab%20Suite%20HTML5%2FSounds%20and%20Sonification%2FMockup%20Videos%2FfelCurrentLoopMockupVideo%2Emp4&referrer=StreamWebApp%2EWeb&referrerScenario=AddressBarCopied%2Eview)

I have a looping sound file [felCurrentLoop.wav](https://o365coloradoedu.sharepoint.com/sites/PHYS-phet-pilot/_layouts/15/stream.aspx?id=%2Fsites%2FPHYS%2Dphet%2Dpilot%2FPhET%20Files%2FPhET%20Files%2FPhET%20Sim%20Design%2FFaraday%27s%20Electromagnetic%20Lab%20Suite%20HTML5%2FSounds%20and%20Sonification%2FSound%20Files%2FfelCurrentLoop%2Ewav&referrer=StreamWebApp%2EWeb&referrerScenario=AddressBarCopied%2Eview). I have it mocked up that if its all the way right the playback rate is 12 semitones higher and as you drag it to the middle its goes from 12 higher to its base playback rate which is 0 higher. 

The sound fades in and fades out as you interact with the sliders similar to other sound mapping. So its only playing with interaction.

The way I have it mocked up is that as you drag it from the middle to the left and the current starts reversing, it plays from +4 semitones in the middle to +16 at max at the far left. This is to differentiate  between the two opposites which will be needed when operating in AC current.

This is complicated for me to explain, so perhaps a short call next week would be best.

**Comment by pixelzoom** [comment]
> Add in looping sound for the current in Electromagnet screen

There is a general implementation that is used for all coils -- electromagnet and pickup -- in all screens.  So I can't add sound for current in one screen or coil without incurring a significant development cost, which may or may not ultimately be needed.   So I'll start by adding this to all coils.  Then we can see if there's a need to have different sounds for different coils, in different screens. 

> This is complicated for me to explain, so perhaps a short call next week would be best.

Yes please, we'll need a Zoom call. It's not clear from the video or description.

FYI, current in the model is referred to as "current amplitude". It is a number in the range [-1,1], where the sign indicates current direction, and 0 is no current. It would be helpful if you could specify how current amplitude should be mapped to sound playback rate.

**Comment by pixelzoom** [comment]
> This is complicated for me to explain, so perhaps a short call next week would be best.

Yes please, we'll need a Zoom call. It's not clear from the video or description.

**Comment by pixelzoom** [comment]
FYI, current in the model is referred to as "current amplitude".  It is a number in the range [-1,1], where the sign indicates current direction, and 0 is no current.  It would be helpful if you could specify how current amplitude should be mapped to sound playback rate.

**Comment by pixelzoom** [comment]
2/13/24 design meeting @arouinfar @Ashton-Morris @emily-phet @pixelzoom

The sound is associated with dragging the voltage slider on the power supply.

Let's do the same thing for AC and DC power supplies.
 
Do this for all Electromagnets, not just the Electromagnet screen.

No sound for pickup coil for now.

The mapping is currentAmplitude => playbackRate

* (0,1] => [0,12]
* 0 = no sound
* [-1,0) => [16,4]

**Comment by pixelzoom** [comment]
2/20/24 design meeting notes (@arouinfar @Ashton-Morris @emily-phet @pixelzoom)

We did not discuss this particular sound further.

On hold until we decide on next steps.

**Comment by pixelzoom** [comment]
Per @kathy-phet and @emily-phet, we will proceed with publishing FEL 1.0 without sound.  So labeling this issues as deferred, to be revisited in a future release. 

I hadn't done any implementation related to this issue yet. So there are no commits, and no related source code or sound files in GitHub.
